MediaBrowser.Controller/Entities/Folder.cs

```diff
@@ -35,6 +36,17 @@ namespace MediaBrowser.Controller.Entities
     /// </summary>
     public class Folder : BaseItem
     {
+        private static Lazy<SemaphoreSlim> _metadataRefreshThrottler = new Lazy<SemaphoreSlim>(() => {
```

Comment:
Should this be placed somewhere else? We're trying to de-bloat the BaseItems.

Reply:
Do you recommend any other spots for it? I assume it can be placed anywhere that `Folder` can access and a place that has access to `ConfigurationManager`

Reply:
I would stick it into `IBaseItemManager/BaseItemManager`

Reply:
Sounds good. Since `Folder` doesn't directly know about `BaseItemManager`, I have routed it through `ProviderManager` now. I think it fits well. Let me know if that's not right.

Reply:
Given that it's static, won't any updates to the config require a restart? We should probably fix that somehow before merging.

Reply:
Good call out. I added handling for that.